src/test/java/bio/terra/service/resourcemanagement/ResourceLockTest.java

```diff
@@ -0,0 +1,193 @@
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(SpringRunner.class)
+@AutoConfigureMockMvc
+@SpringBootTest
+@ActiveProfiles({"google", "connectedtest"})
+@Category(Connected.class)
+public class ResourceLockTest {
+    private final Logger logger = LoggerFactory.getLogger(ResourceLockTest.class);
+
+    @Autowired private LoadDao loadDao;
+    @Autowired private ConfigurationService configService;
+    @Autowired private GoogleResourceConfiguration resourceConfiguration;
+    @Autowired private GoogleResourceService resourceService;
+    @Autowired private ConnectedOperations connectedOperations;
+    @Autowired private GoogleResourceDao resourceDao;
+    @Autowired private DataLocationService dataLocationService;
+    @Autowired private ProfileService profileService;
+    @MockBean private IamService samService;
+
+    private BillingProfileModel profile;
+    private Storage storage;
+
+    @Before
+    public void setup() throws Exception {
+        profile = connectedOperations.createProfileForAccount(resourceConfiguration.getCoreBillingAccount());
+        connectedOperations.stubOutSamCalls(samService);
+        storage = StorageOptions.getDefaultInstance().getService();
+    }
+
+    @After
+    public void teardown() throws Exception {
+        connectedOperations.teardown();
+    }
+
+    @Test
+    // create and delete the bucket, checking that the metadata and cloud state match what is expected
+    public void createAndDeleteBucketTest() {
+        String bucketName = "testbucket_createanddeletebuckettest";
+        String flightId = "testFlightId";
+
+        // create the bucket and metadata
+        GoogleBucketRequest googleBucketRequest = buildBucketRequest(bucketName);
+        GoogleBucketResource bucketResource = resourceService.getOrCreateBucket(googleBucketRequest, flightId);
+
+        checkBucketExists(bucketResource.getResourceId());
+
+        deleteBucket(bucketResource.getName(), bucketResource.getResourceId());
```

Comment:
Maybe put this in teardown so it gets cleaned up if the test fails?

Reply:
Added to teardown. I also kept it in the test, because I find it helpful to walk through all the steps when debugging. I updated the teardown to accommodate already deleted objects.